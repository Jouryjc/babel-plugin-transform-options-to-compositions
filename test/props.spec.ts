import plugin from '../lib/index'
import { transformSync } from '@babel/core'

describe('props -> defineProps', () => {
  test('defineProps', () => {
    const { code } = transformSync(
      `
      export default {
        props: {
          foo: {
            type: String,
            default: 'foo',
          },
          bar: Number,
        },
      }
    `,
      {
        presets: [
          [
            '@babel/preset-env',
            {
              modules: false,
            },
          ],
        ],
        plugins: [plugin],
      }
    )

    expect(code).toMatchSnapshot('props')
  })
})
