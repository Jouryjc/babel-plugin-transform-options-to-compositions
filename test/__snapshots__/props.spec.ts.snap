// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props -> defineProps defineProps: props 1`] = `
"import { defineProps } from 'vue';
const props = defineProps({
  foo: {
    type: String,
    \\"default\\": 'foo'
  },
  bar: Number
});"
`;
